### DISCUSSION TITLE
How can I make the bot only respond to authorized IDs?

### DISCUSSION THREAD

**Discussion by diegogslomp** [aiogram/aiogram#1313] [General]
```python
from aiogram.dispatcher.event.bases import SkipHandler
from aiogram import BaseMiddleware

class AuthMiddleware(BaseMiddleware):
    async def __call__(self, handler, event, data):
        authorized = {"user": 1111}
        if event.from_user.id not in authorized.values():
            raise SkipHandler()
        return await handler(event, data)
```

**Comment by Olegt0rr** [comment]
`401 Unauthorized` from Telegram API Bot means you have bad token, try to issue another one from https://t.me/botfather

**Comment by diegogslomp** [comment]
Thanks for the reply @Olegt0rr! I want the bot to respond only to authorized IDs, so I thought to handle that in a middleware. The code above works I just want to make sure that it's ok to just `return` in case of bad ID.

**Reply by Olegt0rr** [reply]
Sorry, my fault with understanding your request :)

Try to raise `SkipHandler` or `CancelHandler`

**Reply by diegogslomp** [reply]
Snippet updated, thank you!

**Reply by JrooTJunior** [reply]
It works, but you don't need to raise any exception, just return nothing insteead of calling handler.

**Reply by diegogslomp** [reply]
Yes I did in that in the first version and it works, the only difference I notice is that the log is updated with the raise. Thank you!
